src/metatrain/deprecated/pet/default-hypers.yaml

```diff
@@ -1,6 +1,7 @@
 architecture:
 
   name: deprecated.pet
+  version: 1
```

Comment:
I don't think the current version should be defined in the hypers, since this is not something that the user has control over

Reply:
I would rather have it in `Model.__checkpoint_version__`

Reply:
ah from my talk with @PicoCentauri I understood we wanted to do `version` in the input at the same level of `architecture_name`, and not have it in the model class. Of course, I can change this. 
I liked this way becouse when you look at the default_hypers it says: "this is how a input for version X should look like"

Reply:
> I liked this way becouse when you look at the default_hypers it says: "this is how a input for version X should look like"

The version of the checkpoint could also change without changing the version of the hypers, for example if we rename some field internally

Reply:
if I put it in Model.__checkpoint_version__ when I call it  from `model_from_checkpoint` I get the error:

```
AttributeError: type object 'SoapBpnn' has no attribute '__checkpoint_version__'
```

becouse probably no class is initializade 

we could do a method in the class

```
@classmethod
    def get_checkpoint_version(cls) -> int:
        """
        Get the checkpoint version of the model.
        """
        return 1
```

Reply:
for reference the code

```
@classmethod
def get_checkpoint_version(self) -> int:
        """
        Get the checkpoint version of the model.
        """
        return self.__checkpoint_version__
```
get this error
```
TypeError: SoapBpnn.get_checkpoint_version() missing 1 required positional argument: 'self'
```

Reply:
`__checkpoint_version__` needs to be defined on the class, not inside `__init__`. This is what we do for `__supported_devices__` & co already.

Reply:
ahhhhhh I see ok thanks,